lms/djangoapps/letstudy_api/views.py

```diff
@@ -0,0 +1,165 @@
+        Check required parameter is correct.
+        If parameter isn't correct ValueError is raised.
+        :param parameter: object
+        :param parameter_name: string. Parameter's name
+        :param values_list: List of values
+        :return: parameter
+        """
+        if not parameter or (values_list and isinstance(values_list, list) and parameter not in values_list):
+            raise ValueError(self._error_dict[parameter_name].format(value=parameter))
+        return parameter
+
+    def post(self, request):
+        """
+        Create user by the email and the username.
+        """
+        data = dict(request.data.iteritems())
+        data['honor_code'] = "True"
+        data['terms_of_service'] = "True"
+        first_name = request.data.get('first_name', '')
+        last_name = request.data.get('last_name', '')
+
+        try:
+            email = self._check_available_required_params(request.data.get('email'), "email")
+            username = self._check_available_required_params(request.data.get('username'), "username")
+            uid = self._check_available_required_params(request.data.get('uid'), "uid")
+            if check_account_exists(username=username, email=email):
+                return Response(data={"error_message": "User already exists"}, status=status.HTTP_409_CONFLICT)
+            if UserSocialAuth.objects.filter(uid=uid).exists():
+                return Response(
+                        data={"error_message": "Parameter 'uid' isn't unique."},
+                        status=status.HTTP_409_CONFLICT
+                )
+            data['name'] = "{} {}".format(first_name, last_name).strip() if first_name or last_name else username
+            data['first_name'] = first_name
+            data['last_name'] = last_name
+            data['password'] = uuid4().hex
+            for param in ('is_active', 'allow_certificate'):
+                data[param] = data.get(param, True)
+            user = User.objects.create_user(username=username, email=email, password=data['password'])
+            UserProfile.objects.create(user=user)
+            idp_name = OAuth2ProviderConfig.objects.first().backend_name
+            UserSocialAuth.objects.create(user=user, provider=idp_name, uid=uid)
+        except (ValueError, ValidationError) as e:
+            return Response(
+                data={"error_message": e.message},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        return Response(data={'user_id': user.id, 'username': username}, status=status.HTTP_200_OK)
+
+
+class UserEnrollView(APIView, ApiKeyPermissionMixIn):
+    """
+    API endpoint for enrolling user on the course.
+    """
+    authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
+    permission_classes = (ApiKeyHeaderPermission,)
+
+    def post(self, request):
+        """
+        Enroll user on the course.
+
+        :param course: Course id. string, in the format {key type}:{org}+{course}+{run}.
```

Comment:
This param in docstring isn't available in the post method. Please rewrite docstring

Reply:
Done